ui/admin/app/controllers/scopes/scope/session-recordings/index.js

```diff
@@ -4,21 +4,189 @@
+import { debounce } from 'core/decorators/debounce';
+import { inject as service } from '@ember/service';
 
 export default class ScopesScopeSessionRecordingsIndexController extends Controller {
+  // =services
+
+  @service store;
+  @service intl;
+
+  // =attributes
+
+  queryParams = [
+    'search',
+    { times: { type: 'array' } },
+    { users: { type: 'array' } },
+    { scopes: { type: 'array' } },
+    { targets: { type: 'array' } },
+    'page',
+    'pageSize',
+  ];
+
+  @tracked search = '';
+  @tracked times = [];
+  @tracked users = [];
+  @tracked scopes = [];
+  @tracked targets = [];
+  @tracked page = 1;
+  @tracked pageSize = 10;
+
+  /**
+   * Returns object of filters to be used for displaying selected filters
+   * @returns {object}
+   */
+  get filters() {
+    return {
+      allFilters: {
+        times: this.timeOptions,
+        users: this.filterOptions('user'),
+        scopes: this.projectScopes,
+        targets: this.filterOptions('target'),
+      },
+      selectedFilters: {
+        times: this.times,
+        users: this.users,
+        scopes: this.scopes,
+        targets: this.targets,
+      },
+    };
+  }
+
+  /**
+   * Returns array of time options for time filter
+   * @returns {[object]}
+   */
+  get timeOptions() {
+    const now = new Date();
+    const midnight = new Date(now.getFullYear(), now.getMonth(), now.getDate());
+    const threeDaysPast = new Date(midnight);
+    threeDaysPast.setDate(midnight.getDate() - 3);
+    const sevenDaysPast = new Date(midnight);
+    sevenDaysPast.setDate(midnight.getDate() - 7);
+
+    return [
+      {
+        id: midnight.toISOString(),
+        name: this.intl.t('resources.session-recording.filters.time.today'),
+      },
+      {
+        id: threeDaysPast.toISOString(),
+        name: this.intl.t(
+          'resources.session-recording.filters.time.last-three-days',
+        ),
+      },
+      {
+        id: sevenDaysPast.toISOString(),
+        name: this.intl.t(
+          'resources.session-recording.filters.time.last-seven-days',
+        ),
+      },
+    ];
+  }
+
+  /**
+   * Returns unique project scopes from targets
+   * linked to the session recordings
+   * @returns {[object]}
+   */
+  get projectScopes() {
+    const uniqueMap = new Map();
+    this.model.allSessionRecordings.forEach(
+      ({
+        create_time_values: {
+          target: {
+            scope: { id, name, parent_scope_id },
+          },
+        },
+      }) => {
+        if (!uniqueMap.has(id)) {
+          const projectName = name || id;
+          uniqueMap.set(id, { id, name: projectName, parent_scope_id });
+        }
+      },
+    );
+    return Array.from(uniqueMap.values());
+  }
+
+  // =actions
+
+  /**
+   * Looks up org by ID and returns the org name
+   * @param {string} orgID
+   * @returns {string}
+   */
+  @action
+  orgName(orgID) {
+    const org = this.store.peekRecord('scope', orgID);
+    return org.name;
```

Comment:
Should this return `org.displayName` instead ?

Reply:
It should return `org.displayName`. Thanks

Reply:
Fixed.